src/components/organism/HeaderTopNav/index.tsx

```diff
@@ -3,38 +3,57 @@ import { NavLink, useLocation } from "react-router-dom";
 import { POOLS_ROUTE, SWAP_ROUTE } from "../../../app/router/routes.ts";
 import { ReactComponent as AccountImage } from "../../../assets/img/account-image-icon.svg";
 import { ReactComponent as Logo } from "../../../assets/img/logo-icon.svg";
-import { ButtonVariants } from "../../../app/types/enum.ts";
+import { ActionType, ButtonVariants, WalletConnectSteps } from "../../../app/types/enum.ts";
 import { reduceAddress } from "../../../app/util/helper";
 import dotAcpToast from "../../../app/util/toast.ts";
-import { handleConnection, handleDisconnect } from "../../../services/polkadotWalletServices";
+import { handleConnection, handleDisconnect, setTokenBalance } from "../../../services/polkadotWalletServices";
 import { useAppContext } from "../../../state/index.tsx";
 import Button from "../../atom/Button/index.tsx";
 import { t } from "i18next";
 import { useEffect, useState } from "react";
 import { InjectedAccountWithMeta } from "@polkadot/extension-inject/types";
 import { lottieOptions } from "../../../assets/loader/index.tsx";
 import Lottie from "react-lottie";
+import WalletConnectModal from "../WalletConnectModal/index.tsx";
 import LocalStorage from "../../../app/util/localStorage.ts";
+import { ModalStepProps } from "../../../app/types/index.ts";
 
 const HeaderTopNav = () => {
   const { state, dispatch } = useAppContext();
-  const { api, walletConnectLoading } = state;
+  const { walletConnectLoading, api } = state;
   const location = useLocation();
+
   const [walletAccount, setWalletAccount] = useState<InjectedAccountWithMeta>({} as InjectedAccountWithMeta);
+  const [modalStep, setModalStep] = useState<ModalStepProps>({ step: WalletConnectSteps.stepExtensions });
+  const [walletConnectOpen, setWalletConnectOpen] = useState(false);
 
   const walletConnected = LocalStorage.get("wallet-connected");
 
   const connectWallet = async () => {
     try {
-      await handleConnection(dispatch, api);
+      dispatch({ type: ActionType.SET_WALLET_CONNECT_LOADING, payload: true });
+      setWalletConnectOpen(true);
```

Comment:
This should be outside of the try block 
dispatch({ type: ActionType.SET_WALLET_CONNECT_LOADING, payload: true });
setWalletConnectOpen(true);